spec.bs

```diff
@@ -231,6 +235,100 @@ Requests to key commitment endpoints should result in a JSON response
 All field names and their values are strings. When new key commitments are
 fetched for an issuer, previous commitments are discarded.
 
+Algorithms {#algorithms}
+====================================
+
+To <dfn>determine whether associating an issuer would exceed the top-level limit</dfn> given an [=/origin=] |issuer| and an [=/origin=] |topLevel|, run the following steps:
+
+1. Let |limit| be 2.
+2. TODO
+
+To <dfn>associate the issuer</dfn> |issuer| (an [=/origin=]) with the [=/origin=] |topLevel|, run the following steps:
+
+1. TODO
+
+To <dfn>look up the key commitments</dfn> for a given [=/origin=] |issuer| and an [=/origin=] |topLevel|, run the following steps:
+
+1. TODO
+
+To determine whether an [=/origin=] |issuer| <dfn>is associated with</dfn> a given [=/origin=] |topLevel|, run the following steps:
+
+1. TODO
+
+To <dfn>retrieve a token</dfn> for an [=/origin=] |issuer|, run the following steps:
+
+1. TODO
+
+To <dfn>discard tokens</dfn> from an [=/origin=] |issuer|, run the following steps:
+
+1. TODO
+
+To get the <dfn>number of tokens</dfn> from an [=/origin=] |issuer|, run the following steps:
+
+1. TODO
+
+To get the <dfn>max batch size</dfn> for an [=/origin=] |issuer|, run the following steps:
+
+1. TODO
+
+To <dfn>generate blinded tokens</dfn> given a number |numTokens|, run the following steps:
```

Comment:
non-blocking nit: I think we want to say "masked" instead of "blinded" - I can follow an issue, since this isn't the only occurance.

Reply:
Let's follow up in an issue here